extra/checksums/multi/multi.factor

```diff
@@ -0,0 +1,31 @@
+! Copyright (C) 2018 Alexander Ilin.
+! See http://factorcode.org/license.txt for BSD license.
+USING: accessors checksums checksums.common destructors fry kernel
+sequences ;
+IN: checksums.multi
+
+TUPLE: multi-checksum list ;
+INSTANCE: multi-checksum block-checksum
+C: <multi-checksum> multi-checksum
+
+TUPLE: multi-state < disposable states results ;
+
+M: multi-checksum initialize-checksum-state
+    list>> [ initialize-checksum-state ] V{ } map-as
+    multi-state new-disposable swap >>states ;
+
+M: multi-state dispose*
+    states>> [ [ dispose ] each ] when* ;
+
+M: multi-state add-checksum-bytes
+    '[ [ _ add-checksum-bytes ] map! ] change-states ;
```

Comment:
Not necessary to ``change-states``... since you're using ``map!``.

```[ states>> ] keep '[ _ add-checksum-bytes ] map! drop``?

Reply:
We need to keep the object instance on the stack, so it's actually
```
    [ dup states>> ] dip '[ _ add-checksum-bytes ] map! drop ;
```
vs.
```
    '[ [ _ add-checksum-bytes ] map! ] change-states ;
```
And the latter wins the Code Golf, am I right?

Reply:
Hah, sure I'm just saying since you're mapping in place you don't need the "set states".  But sure, it's not really a performance issue.